uint32 x_position
uint32 y_position
uint32 width
uint32 height
float32 distance
uint8 color_index
uint8 object_type
